docs/proposals/web-hook.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Dynamic Admission Control
+status: Alpha
+authors:
+    - "@chendave"
+approvers:
+  - rohitsardesai83
+  - fisherxu
+  - TBD
+creation-date: 2019-07-19
+last-updated: 2019-07-20
+---
+
+# Motivation
+As the evolving of the project, it is foreseeable more API and resouce will be added in the project. kubeedge so far lack of effective way of pre-processing for the object configuration, for example, whether the pod that is going to be created contains the unwanted label, should the specific configmap be protected from deletion etc.
+
+There is a concrete example on the github [issue 845](https://github.com/kubeedge/kubeedge/issues/845), the issue there cannot be addressed by [CRD validation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation), so we need to explore other approach to achieve that purpose.
+
+
+## Goals
+* A framework to enable admission control in kubeedge.
+* Address the edge cases where kubernetes CRD validation cannot accomplish.
+* Create basic integration testcases.
+
+## Non-goals
+* Basic CRD validation should be done by [validation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation).
+* Start from `Device` and `DeviceModel` other kinds of resource will be evaluated later and will not be included in the first alpha version.
+
+# Proposal
+Propose using Kubernetes [Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers) to determine whether to accept or deny the request, the decision is based on how is the policy is configured, this give us chance to validate the request before persisting the object.
+
+
+# Design Details
+
+## Admission controller on control-plane
+
+Create a new admission control when the module `controller` is started.
+```go
+func (ctl *Controller) Start(c *bcontext.Context) {
```

Comment:
The admission controller is not associated with cloud part(cloudhub/controller), it should be started independently.
Where the code base placed， let's disscuss it, how do you think? @chendave @kevin-wangzefeng @rohitsardesai83 @sids-b

Reply:
Do you mean we need another module beside "cloudhub", "controller", and "devicecontroller"?

Reply:
is that too heavy?